crates/pixi_build_types/src/procedures/conda_outputs.rs

```diff
@@ -0,0 +1,213 @@
+//! Describes the `conda/outputs` request and its parameters.
+//!
+//! This request is used to compute all the outputs that a particular backend
+//! can provide. It returns the identifiable metadata of the outputs, including
+//! the dependencies required to be able to build them.
+//!
+//! This API was introduced in Pixi Build API version 1.
+
+use ordermap::OrderSet;
+use rattler_conda_types::{NoArchType, PackageName, Platform, VersionWithSource};
+use serde::{Deserialize, Serialize};
+use serde_with::serde_as;
+use std::{
+    collections::{BTreeSet, HashMap},
+    path::PathBuf,
+};
+
+use crate::{BinaryPackageSpecV1, PackageSpecV1, project_model::NamedSpecV1};
+
+pub const METHOD_NAME: &str = "conda/outputs";
+
+/// Parameters for the `conda/outputs` request.
+///
+/// The result of this request should be a list of packages that can be built by
+/// this particular source dependency.
+#[derive(Debug, Serialize, Deserialize, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct CondaOutputsParams {
+    /// The native platform for which the outputs should be computed.
+    ///
+    /// This is usually the same platform as the platform on which the backend
+    /// is running but when cross-compiling this could be different.
+    pub host_platform: Platform,
+
+    /// The possible variants by the pixi workspace.
+    pub variant_configuration: Option<HashMap<String, Vec<String>>>,
+
+    /// A directory that can be used by the backend to store files for
+    /// subsequent requests. This directory is unique for each separate source
+    /// dependency.
+    ///
+    /// The directory may not yet exist.
+    pub work_directory: PathBuf,
+}
+
+/// Contains the result of the `conda/outputs` request.
+#[derive(Debug, Serialize, Deserialize, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct CondaOutputsResult {
+    /// Metadata of all the packages that can be built.
+    pub outputs: Vec<CondaOutputMetadata>,
+
+    /// The files that were read as part of the computation. These files are
+    /// hashed and stored in the lock-file. If the files change, the
+    /// lock-file will be invalidated.
+    pub input_globs: BTreeSet<String>,
+}
+
+#[serde_as]
+#[derive(Debug, Serialize, Deserialize, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct CondaOutputMetadata {
+    /// The identifier of the output.
+    pub identifier: CondaOutputIdentifier,
```

Comment:
maybe we can rename it to `metadata `? I would expect identifier to be an id or something, but here it is more like a subset of some metadata kind